frontend/ui/widget/multiinputdialog.lua

```diff
@@ -249,4 +265,24 @@ function MultiInputDialog:onKeyboardHeightChanged()
     UIManager:setDirty("all", "flashui")
 end
 
+function InputDialog:addWidget(widget, re_init)
+    table.insert(self.layout, #self.layout, {widget})
+    if not re_init then -- backup widget for re-init
+        widget = CenterContainer:new{
+            dimen = Geom:new{
+                w = self.width,
+                h = widget:getSize().h,
+            },
+            widget,
+        }
+        if not self._added_widgets then
+            self._added_widgets = {}
+        end
+        table.insert(self._added_widgets, widget)
+    end
+    -- insert widget before the bottom buttons
```

Comment:
The origina you copied from inputdialog.lua has:
`-- insert widget before the bottom buttons and their previous vspan`
Not exactly the case in here ? If it is, please just keep them exactly similar (so we don't have to wonder if touching this later).

Reply:
Fixed (it is indeed directly analogous)